src/tests/functional/plugin/conformance/test_runner/conformance_infra/include/conformance.hpp

```diff
@@ -44,19 +46,49 @@ inline ov::AnyMap readPluginConfig(const std::string &configFilePath) {
     return config;
 }
 
-inline std::vector<std::string> getModelPaths(const std::vector<std::string>& conformance_ir_paths) {
+inline std::vector<std::string> getModelPaths(const std::vector<std::string>& conformance_ir_paths,
+                                              const std::string opName = "") {
+    static std::list<std::string> dirList;
+    static bool listPrepared = false;
+    if (!listPrepared) {
+        // Looking for any applicable files in a folders
+        for (const auto& conformance_ir_path : conformance_ir_paths) {
+            std::vector<std::string> tmp_buf;
+            if (CommonTestUtils::directoryExists(conformance_ir_path)) {
+                tmp_buf =
+                    CommonTestUtils::getFileListByPatternRecursive({conformance_ir_path}, {std::regex(R"(.*\.xml)")});
+            } else if (CommonTestUtils::fileExists(conformance_ir_path)) {
+                tmp_buf = CommonTestUtils::readListFiles({conformance_ir_path});
+            } else {
+                continue;
+            }
+            //Save it in a list
+            dirList.insert(dirList.end(), tmp_buf.begin(), tmp_buf.end());
+        }
+        listPrepared = true;
+    }
+
     std::vector<std::string> result;
-    for (const auto& conformance_ir_path : conformance_ir_paths) {
-        std::vector<std::string> tmp_buf;
-        if (CommonTestUtils::directoryExists(conformance_ir_path)) {
-            tmp_buf = CommonTestUtils::getFileListByPatternRecursive({conformance_ir_path}, {std::regex(R"(.*\.xml)")});
-        } else if (CommonTestUtils::fileExists(conformance_ir_path)) {
-            tmp_buf = CommonTestUtils::readListFiles({conformance_ir_path});
-        } else {
-            continue;
+
+    if (opName != "" && opName != CONFORMANCE_OTHER_OPS) {
+        // Looking for files which contains /opName/ in file path, an expecting file structure is: /opName/precision/file.xml
+        auto opLookup = std::regex("[\\\\/]" + opName + "-?([^\\\\/])?[\\\\/]", std::regex::icase);
+        auto it = dirList.begin();
+        while (it != dirList.end()) {
```

Comment:
MINOR: taste by taste: `for(const auto& dir : dirList)`?

Reply:
np